```
PS C:\Users\Cezar\kraftkit> go test ./initrd
--- FAIL: TestNewFromDirectory (0.03s)
    directory_test.go:83: Encountered unexpected file in cpio archive: testdata/rootfs
    directory_test.go:94: file [/entrypoint.sh]: got size 27, expected 25
    directory_test.go:94: file [/etc/app.conf]: got size 18, expected 16
    directory_test.go:88: file [/lib/libtest.so.1]: got mode 0100000, expected 0120000
    directory_test.go:91: file [/lib/libtest.so.1]: got linkname "", expected "libtest.so.1.0.0"
    directory_test.go:94: file [/lib/libtest.so.1]: got size 16, expected 0
    directory_test.go:94: file [/lib/libtest.so.1.0.0]: got size 10, expected 9
```

@antoineco The tests still fail on windows but now I think this is because of something else:
- It appends `\CR` when reading files, that's why each line is an extra line
- Links behave differently within Windows

--- a/initrd/directory.go
+++ b/initrd/directory.go
@@ -81,6 +81,7 @@ func (initrd *directory) Build(ctx context.Context) (string, error) {
 		}
 
 		internal := strings.TrimPrefix(path, filepath.Clean(initrd.path))
+		internal = filepath.ToSlash(internal)
 		if internal == "" {
 			return nil // Do not archive empty paths
 		}

--- a/initrd/directory.go
+++ b/initrd/directory.go
@@ -55,10 +55,14 @@ func (initrd *directory) Build(ctx context.Context) (string, error) {
 	if initrd.opts.output == "" {
 		fi, err := os.CreateTemp("", "")
 		if err != nil {
-			return "", fmt.Errorf("could not make temporary directory: %w", err)
+			return "", fmt.Errorf("could not make temporary file: %w", err)
 		}
 
 		initrd.opts.output = fi.Name()
+		err = fi.Close()
+		if err != nil {
+			return "", fmt.Errorf("could not close temporary file: %w", err)
+		}
 	}
 
 	f, err := os.OpenFile(initrd.opts.output, os.O_RDWR|os.O_CREATE, 0o644)

